File: GFG/Recursion/palindromecheck.py
def check(asd):
    n= len(asd)
    if (n==0 or n==1):
        return True
    if (asd[0]==asd[-1]):
        return check(asd[1:-1])
    else: 
        return False

x = input()
print(check(x))

File: GFG/Recursion/palindromecheck.cpp
#include<iostream>
#include<string>
using namespace std;

bool check(string asd){
    int n = asd.length();
    if (n==1 or n==1){
        return true;
    }
    if (asd[0]==asd[n-1]){
        return true;
    }
    else{
        return false;
    }
    return check(asd.substr(1, n - 2));
}

int main(){
    string x;
    cin>>x;
    cout<<check(x);
    return 0;
}